src/urllib3/connectionpool.py

```diff
@@ -1091,7 +1092,22 @@ def _validate_conn(self, conn: BaseHTTPConnection) -> None:
         if conn.is_closed:
             conn.connect()
 
-        if not conn.is_verified:
+        if (
+            conn.proxy
+            and conn.proxy.scheme == "https"
+            and conn.proxy_is_verified is False
+        ):
+            warnings.warn(
+                (
+                    f"Unverified HTTPS request is being made using proxy host '{conn.proxy.host}'. "
+                    "Adding certificate verification is strongly advised. See: "
+                    "https://urllib3.readthedocs.io/en/latest/advanced-usage.html"
+                    "#tls-warnings"
+                ),
+                InsecureProxyWarning,
+            )
+
+        elif not conn.is_verified:
```

Comment:
Since both can be true, can we change this `elif` to an `if` and add a testcase for both warnings being emitted in one connection?

Reply:
There seems to be something off when I am using a proxy. The following snippet will raise both `InsecureRequestWarning` and `InsecureProxyWarning` after I change the *elif to if*. 

```
import urllib3
proxy = urrlib3.ProxyManager('https://localhost:8443', cert_reqs='NONE')
res = proxy.request('GET', 'https://www.google.com')
```
Outputs the following. 

```
/home/tushar55/projects/urllib3/src/urllib3/connectionpool.py:1100: InsecureProxyWarning: Unverified HTTPS request is being made using proxy host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
/home/tushar55/projects/urllib3/src/urllib3/connectionpool.py:1111: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(

```

Both the conn.host and conn.proxy.host is set to the same value.  `conn._tunnel_host` is set to `www.google.com` 

I suspect this line is resetting the https://github.com/urllib3/urllib3/blob/main/src/urllib3/connectionpool.py#L1062 updates the host and port when proxy is being used. What is the intended behaviour here ?

Reply:
A few more examples that I tested out locally after setting up a simple python http server on port 8000 using `python -m http.server`

```
import urllib3
proxy = urllib3.ProxyManager('https://localhost:8443', ca_certs='/home/tushar55/projects/urllib3/dummyserver/certs/cacert.pem', cert_reqs='NONE')
res = proxy.request('GET', 'http://127.0.0.1:8000')
print(res.status, res.data)
```

This raises the following error - 
```
/home/tushar55/projects/urllib3/src/urllib3/connectionpool.py:1146: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhos
t'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
```

Reply:
This is definitely confusing, but in the case of an HTTPS origin domain we use CONNECT tunneling so the actual destination host we want to connect to is `HTTPConnection._tunnel_host`. There might need to be some modifications to a warning message in order to pick the right host to use. Hopefully this helps?

Reply:
Yeah I think it's clear now. So, if certificate verification is disabled for a proxy manager then it should throw both the `InsecureProxy` and `InsecureRequest` warnings, right?

```
proxy = urllib3.ProxyManager('https://localhost:8443', cert_reqs='NONE')
res = proxy.request('GET', 'https://localhost:8000')
```

and for normal PoolManger, it should just throw `InsecureRequest` warning. 

```
pool = urllib3.PoolManager(cert_reqs='NONE')
res = pool.request('GET', 'https://localhost:8000')
```

I will update the `InsecureRequest` warning's message to `host` or `_tunnel_host` depending upon whether we are using a proxy or not.